cpp/src/arrow/python/decimal.cc

```diff
@@ -214,12 +214,14 @@ DecimalMetadata::DecimalMetadata(int32_t precision, int32_t scale)
     : precision_(precision), scale_(scale) {}
 
 Status DecimalMetadata::Update(int32_t suggested_precision, int32_t suggested_scale) {
-  const int32_t current_precision = precision_;
-  precision_ = std::max(current_precision, suggested_precision);
-
   const int32_t current_scale = scale_;
   scale_ = std::max(current_scale, suggested_scale);
 
+  const int32_t current_precision = precision_;
+  auto n_integral = std::max(std::max(current_precision, 0) - std::max(current_scale, 0),
```

Comment:
Those additional `max(current, 0)` checks make it a bit ugly, but this is required (I think) because on the first call, the current_precision or current_scale can still be the int32 min value (-2147483648)

Reply:
If `current_precision` is int32:min, looks `current_scale` must also be int32::min? So without std::max, guess it should also work.
But I think keep std::max there is better in case we change the behaviour.